src/inverted-index.js

```diff
@@ -1,95 +1,229 @@
+	 *  takes a string and returns an object with the book position of the string
+	 *
+	 * @param {String} terms
+	 * @param {String} filepath
+	 * @returns {Object}
+	 *
+	 *
+	 */
+
+  searchIndex(terms, filepath) {
+    let termsArr;
+    if (Array.isArray(terms)) {
+      terms = terms.join(" ");
+    }
+    termsArr = Index.removePunctuation(terms);
+    let subResult = {};
+    if (filepath) {
+      for (let index in termsArr) {
+        if (this.files[filepath].index[termsArr[index]]) {
+          subResult[termsArr[index]] = this.files[filepath].index[termsArr[index]];
+        }
+        else {
+          subResult[termsArr[index]] = [];
+        }
+      }
+      return subResult;
+    }
+    return this.searchAll(terms, this.files);
+  }
+
+	/**
+	 * Search all
+	 *
+	 * If particular file is not selected, it returns search results from all
+	 * files
+	 *
+	 * @param {String} terms
+	 * @returns {Object}
+	 *
+	 *
+	 */
+
+  searchAll(terms) {
+    let termsArr = Index.removePunctuation(terms);
+    if (!this.files.index) {
+      this.collateBooks();
+    }
+    let allIndex = this.files.index;
+    let subResult = {};
+    for (let index in termsArr) {
+      for (let indexedWord in allIndex) {
+        if (termsArr[index] === indexedWord) {
+          subResult[termsArr[index]] = allIndex[indexedWord];
+        }
+      }
+      if (!subResult[termsArr[index]]) {
+        subResult[termsArr[index]] = [];
+      }
+    }
+    return subResult;
+  }
+
+	/** Get index
+	 *
+	 * takes filename and returns index for that particular file
+	 *
+	 * @param {String} filename
+	 * @returns {Object}
+	 *
+	 *
+	 */
+  getIndex(filename) {
+    filename = filename || false;
+    if (filename) {
+      return this.files[filename].index;
+    }
+  }
+
+  /** Get all Books
+     *
+     * sets key with value containing all the books
+     * @returns {Object}
+     *
+     *
+     */
+  getAllBooks() {
+    let booksall = [];
+    for (let filename in this.files) {
+      if (filename !== 'allBooks' && filename !== 'index') {
+        booksall = booksall.concat(this.files[filename].books);
+      }
+    }
+    this.files['allBooks'] = booksall;
+  }
+
+
+
+  collateBooks() {
+    this.getAllBooks();
+    this.createIndex(null, this.files);
+  }
+
+  /** Get all Index
+     *
+     * returns index of all books in all files
+     * @returns {Object}
+     *
+     *
+     */
+  getAllIndex() {
+    return this.files.index;
+  }
+
+  /** Get all Index
+     *
+     * Checks if contents of file uploaded is in the correct JSON
+     * format with a title and text property returns true or false
+     *
+     * @param {String} uploadedFile
+     * @returns {bool}
+     *
+     *
+     */
+  isValidJSON(uploadedFile) {
+    let isValid = true;
+    let file;
+    try {
+      file = JSON.parse(uploadedFile);
+      if (Array.isArray(file)) {
+        file.forEach((item) => {
+          if (!item.hasOwnProperty('title') || !item.hasOwnProperty('text')) {
```

Comment:
Use the AND instead of OR operator such that both conditions are true and not either of the conditions.